app/helpers/messages_helper.rb

```diff
@@ -0,0 +1,26 @@
+module MessagesHelper
```

Comment:
Would like to wrap this in some kind of PORO class, similar to my comment on vouchers controller.

I believe all helpers are loaded for all views, and they're intended for "decorator"-type "logic". For example, formatting dates. Here we're (1) parsing the response, (2) hitting the database for more data, (3) performing logic to figure out what string to return. This is all 3 MVC layers in one helper object!

Reply:
decent explanation of separation of concerns within an MVC app:
https://staceymck.medium.com/an-intro-to-partials-and-helpers-in-ruby-on-rails-10d62d85da24